### PR TITLE
Add shared example for client data objects

### PR DESCRIPTION
**Pull request by jessieay**
* Shared examples have tests for shared behavior that was previously mostly untested

### CODE REVIEW

File: app/models/proposal.rb [hunk 1]
```diff
@@ -119,7 +115,7 @@ def existing_observation_for(user)
   end
 
   def eligible_observers
-    if observations.count > 0
+    if observations.any?
```
Comment (pkarman) [inline app/models/proposal.rb]: afaik this results is different sql and may be impacted by object caching.

`observations.count` will execute a SQL `select count(*) from ....` while `observations.any?` loads the objects (if not already) and counts them.

The former is more accurate and efficient, imo.

**Comment by pkarman** [comment]
@jessieay I like the unified testing of client code. Just one comment about a refactor that has subtly different effects.

File: app/models/proposal.rb (hunk 1 above)
Comment (jessieay) [inline app/models/proposal.rb]: ah ok! Will change back

Comment (jessieay) [inline app/models/proposal.rb]: updated at 09e3d4b935a6eda4382f865522c05c76433487b5

**Comment by pkarman** [comment]
:+1: 

assuming this is all regression testing, so I will QA when it goes green.

**Comment by jessieay** [comment]
I re-ran the build since it was an unrelated failure and now it is green. worrisome? I haven't seen too many intermittently failing specs...

Yes, regression testing indeed.

**Comment by pkarman** [comment]
QA looks good on c2-dev

**Pull request merged**